libcudacxx/include/cuda/std/detail/libcxx/include/__config

```diff
@@ -76,21 +76,39 @@
 #define _LIBCUDACXX_CUDACC_BELOW_12_3
 #endif // defined(_LIBCUDACXX_CUDACC) && _LIBCUDACXX_CUDACC_VER < 1203000
 
+// Ensure that a user can disable warnings coming from our headers
+#ifndef _CCCL_IMPLICIT_SYSTEM_HEADER
+#  if defined(_LIBCUDACXX_COMPILER_GCC)
+#    define _CCCL_IMPLICIT_SYSTEM_HEADER _Pragma("GCC system_header")
+#  elif defined(_LIBCUDACXX_COMPILER_CLANG)
+#    define _CCCL_IMPLICIT_SYSTEM_HEADER _Pragma("clang system_header")
+#  elif defined(_LIBCUDACXX_COMPILER_MSVC)
+#    define _CCCL_IMPLICIT_SYSTEM_HEADER __pragma(system_header)
+#  elif defined(_LIBCUDACXX_COMPILER_NVHPC)
+#    define _CCCL_IMPLICIT_SYSTEM_HEADER _Pragma("GCC system_header")
```

Comment:
@dkolsen-pgi my initial testing showed that this works for nvc++ is that the right approach?

Reply:
That appears to be the correct approach for `nvc++`.  I am not familiar with that pragma, but in my testing it seems to work correctly.